ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -406,14 +406,61 @@ def _router_bgp_vlans_vlan(self, vlan, tenant, vrf) -> dict | None:
     def _evpn_vlan_aware_bundles(self) -> bool:
         return get(self._hostvars, "evpn_vlan_aware_bundles", default=False)
 
-    def _get_vlan_aware_bundle_name_tuple(self, vlan: dict) -> tuple[str, bool] | None:
+    def _get_vlan_aware_bundle_name_tuple_for_l2vlans(self, vlan: dict) -> tuple[str, bool] | None:
         """
         Return a tuple with string with the vlan-aware-bundle name for one VLAN and a boolean saying if this is a evpn_vlan_bundle.
         """
         if vlan.get("evpn_vlan_bundle") is not None:
             return (str(vlan.get("evpn_vlan_bundle")), True)
         return (str(vlan.get("name")), False)
 
+    def _get_vlan_aware_bundle_name_tuple_for_svis(self, vlan: dict) -> tuple[str, bool] | None:
+        """
+        Return a tuple with string with the vlan-aware-bundle name for one VLAN and a boolean saying if this is a evpn_vlan_bundle.
+        If no bundle is configured, it will return an empty string as name, since the calling function will then get all svis without bundle
+        grouped under "".
+        """
+        if vlan.get("evpn_vlan_bundle") is not None:
+            return (str(vlan.get("evpn_vlan_bundle")), True)
+        return ("", False)
+
+    def _get_evpn_vlan_bundle(self, vlan: dict, bundle_name: str) -> dict:
+        """
+        Return an evpn_vlan_bundle dict if it exists, else raise an exception.
+        """
+        if (evpn_vlan_bundle := get_item(self._hostvars["evpn_vlan_bundles"], "name", bundle_name)) is None:
```

Comment:
`self._hostvars["evpn_vlan_bundles"]` will throw a `KeyError` if `evpn_vlan_bundles` does not exist. Should we handle this as well and include it in the `raise AristaAvdMissingVariableError` message below?

Reply:
Good catch. Thanks! I will fix this.

Reply:
Fixed in https://github.com/aristanetworks/avd/pull/3623/commits/ed5bdd66df71aed06d8963386f34f191d9021abd